test/integration/trigger.test.js

```diff
@@ -0,0 +1,202 @@
+'use strict';
+
+var chai = require('chai')
+  , Sequelize = require('../../index')
+  , expect = chai.expect
+  , Support = require(__dirname + '/../support')
+  , dialect = Support.getTestDialect()
+  , current = Support.sequelize;
+
+
+if(current.dialect.supports.tmpTableTrigger)
+{
+  describe(Support.getTestDialectTeaser('Model'), function() {
+    describe('trigger', function() {
+
+
+        it('should return output rows after insert', function() {
+          var self = this;
+
+          var triggerQuery = 'create trigger User_ChangeTracking on [users] for insert,update, delete \n' +
+                              'as\n' +
+                                'SET NOCOUNT ON\n' +
+                                'if exists(select 1 from inserted)\n' +
+                                'begin\n' +
+                                  'select * from inserted\n' +
+                                'end\n' +
+                                'if exists(select 1 from deleted)\n' +
+                                'begin\n' +
+                                  'select * from deleted\n' +
+                                'end\n';
+
+          var User = self.sequelize.define('user', {
+            username: {
+              type: Sequelize.STRING,
+              field:'user_name'
+            }
+          },{
+            hasTrigger:true
+          });
+
+          return User.sync({force: true}).then(function () {
```

Comment:
`return User.sync({force: true}).bind(this).then(function () {`

Allows you to use `this` inside the promise chain, and remove `var self